lib/api/controllers/adminController.js

```diff
@@ -101,35 +119,64 @@ class AdminController {
       .then(() => this.kuzzle.internalEngine.bootstrap.createDefaultRoles());
 
     if (refresh === 'wait_for') {
-      return promise.then(() => result);
+      return promise.then(() => {
+        delete _locks.resetSecurity;
+
+        return result;
+      });
     }
 
+    delete _locks.resetSecurity;
     return Bluebird.resolve({ acknowledge: true });
   }
 
   /**
    * Reset all indexes created by users
    */
-  resetDatabase () {
-    const indexes = Object.keys(this.kuzzle.indexCache.indexes).filter(idx => idx[0] !== '%');
+  resetDatabase (request) {
+    if (_locks.resetDatabase) {
+      throw new BadRequestError('Kuzzle is already reseting all indexes');
+    }
+    _locks.resetDatabase = true;
 
-    Bluebird.map(indexes, index => {
-      const request = new Request({ index });
+    const
+      indexes = Object.keys(this.kuzzle.indexCache.indexes).filter(idx => idx[0] !== '%'),
+      refresh = request.input.args.refresh;
 
-      return this.kuzzle.services.list.storageEngine.deleteIndex(request)
-        .then(() => delete this.kuzzle.indexCache.indexes[index]);
+    const promise = Bluebird.map(indexes, index => {
+      const req = new Request({ index });
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "request" hides or potentially hides a variable declared in an outer scope at line 136. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)